File: domain/src/main/java/com/ortin/gesturetranslator/domain/repository/RecognizeImageRepository.java
package com.ortin.gesturetranslator.domain.repository;

import androidx.annotation.NonNull;

import com.ortin.gesturetranslator.domain.listeners.RecognizeImageListener;
import com.ortin.gesturetranslator.domain.models.Image;

public interface RecognizeImageRepository {
    void recogniseImage(@NonNull Image image);
    void setRecogniseListener(@NonNull RecognizeImageListener recognizeImageListener);
}

File: domain/src/main/kotlin/com/ortin/gesturetranslator/domain/repository/RecognizeImageRepository.kt
package com.ortin.gesturetranslator.domain.repository

import com.ortin.gesturetranslator.domain.listeners.RecognizeImageListener
import com.ortin.gesturetranslator.domain.models.Image

interface RecognizeImageRepository {
    fun recogniseImage(image: Image)

    fun setRecogniseListener(recognizeImageListener: RecognizeImageListener)
}
